CONTRIBUTING.md

```diff
@@ -61,20 +61,62 @@ Before sending us a pull request, please ensure that:
 1. Install test dependencies, including coverage, using `pip install ".[test]"`
 1. Run the following tox command and verify that all code checks and unit tests pass: `tox tests/unit`
 
-You can also run a single test with the following command: `tox -e py36 -- -s -vv <path_to_file><file_name>::<test_function_name>`  
+You can also run a single test with the following command: `tox -e py36 -- -s -vv <path_to_file><file_name>::<test_function_name>`
   * Note that the coverage test will fail if you only run a single test, so make sure to surround the command with `export IGNORE_COVERAGE=-` and `unset IGNORE_COVERAGE`
   * Example: `export IGNORE_COVERAGE=- ; tox -e py36 -- -s -vv tests/unit/test_sagemaker_steps.py::test_training_step_creation_with_model ; unset IGNORE_COVERAGE`
 
 
 ### Running the Integration Tests
 
-Our CI system runs integration tests (the ones in the `tests/integ` directory), in parallel, for every Pull Request.  
-You should only worry about manually running any new integration tests that you write, or integration tests that test an area of code that you've modified.  
+Our CI system runs integration tests (the ones in the `tests/integ` directory), in parallel, for every Pull Request.
+You should only worry about manually running any new integration tests that you write, or integration tests that test an area of code that you've modified.
+#### Setup
 
-1. Follow the instructions at [Set Up the AWS Command Line Interface (AWS CLI)](https://docs.aws.amazon.com/polly/latest/dg/setup-aws-cli.html).
+If you haven't done so already, install tox and test dependencies:
+1. `pip install tox`
+1. `pip install .[test]`
+
+#### AWS Credentials
+Follow the instructions at [Set Up the AWS Command Line Interface (AWS CLI)](https://docs.aws.amazon.com/polly/latest/dg/setup-aws-cli.html).
+#### Create IAM Roles
+
+The tests use two IAM roles to give Step Functions and SageMaker permissions to access AWS resources in your account. Use the following commands in the root directory of this repository:
+
+```bash
+aws iam create-role \
+   --role-name StepFunctionsMLWorkflowExecutionFullAccess \
+   --assume-role-policy-document file://tests/integ/resources/StepFunctionsMLWorkflowExecutionFullAccess-TrustPolicy.json
+```
+
+```bash
+aws iam put-role-policy \
+   --role-name StepFunctionsMLWorkflowExecutionFullAccess \
+   --policy-name StepFunctionsMLWorkflowExecutionFullAccess \
+   --policy-document file://tests/integ/resources/StepFunctionsMLWorkflowExecutionFullAccess-Policy.json
+```
+
+```bash
+aws iam attach-role-policy \
+   --role-name StepFunctionsMLWorkflowExecutionFullAccess \
+   --policy-arn arn:aws:iam::aws:policy/CloudWatchEventsFullAccess
```

Comment:
The tests probably don't need full access to all EventBridge APIs. If possible, let's scope it down to the minimal set. 

The failures I saw were in creating managed rules. it might be the `Resource` that's missing an ARN. Scoping it to `"Resource": "arn:aws:events:*:*:rule/StepFunctionsGetEventsFor*"` might be a good way to scope permissions down to the rule naming convention Step Functions uses without having to update the policy each time.

https://github.com/aws/aws-step-functions-data-science-sdk-python/blob/42ac3fc1b2675b4414a83d7ac1cddbd98ab7fec4/tests/integ/resources/StepFunctionsMLWorkflowExecutionFullAccess-Policy.json#L59-L73

Reply:
Makes sense, we do not want to five full access if not needed! I will remove full access to EventBridge and target which resource arn is missing.

Reply:
I removed the `CloudWatchEventsFullAccess` policy from `StepFunctionsMLWorkflowExecutionFullAccess` and was able to run `tox tests/integ` successfully. Will remove the instructions in the next commit!

Reply:
Confirming that the EventBridge permissions in `StepFunctionsMLWorkflowExecutionFullAccess` are sufficient to run the integ tests. My previous test run was failing to create the `StepFunctionsGetEventsForSageMakerTrainingJobsRule` rule because I had some leftover artifacts from a preprod stage in my account. They succeeded after deleting the rule and rerunning the tests.